unicodetools/src/main/java/org/unicode/text/UCD/VersionedSymbolTable.java

```diff
@@ -0,0 +1,426 @@
+     * binary-query-expression with a property-value where the queried property is General_Category.
+     */
+    private UnicodeSet getGeneralCategorySet(IndexUnicodeProperties iup, String propertyValue) {
+        var gc = iup.getProperty(UcdProperty.General_Category);
+        for (var entry : COARSE_GENERAL_CATEGORIES.entrySet()) {
+            final var aliases = entry.getKey().getNames().getAllNames();
+            if (aliases.stream().anyMatch(a -> UnicodeProperty.equalNames(propertyValue, a))) {
+                UnicodeSet result = new UnicodeSet();
+                for (var value : entry.getValue()) {
+                    gc.getSet(value, result);
+                }
+                return result;
+            }
+        }
+        return gc.getSet(propertyValue);
+    }
+
+    @Override
+    public boolean applyPropertyAlias(String beforeEquals, String afterEquals, UnicodeSet result) {
+        result.clear();
+        String leftHandSide = beforeEquals;
+        String propertyPredicate = afterEquals;
+        boolean interiorlyNegated = false;
+        int posNotEqual = beforeEquals.indexOf('≠');
+        // TODO(egg): We cannot distinguish \p{X=} from \p{X} in this API, both give us an empty
+        // string as afterEquals.  This is an @internal API, so we could change it to pass null in
+        // the unary case.
+        if (posNotEqual >= 0) {
+            propertyPredicate =
+                    afterEquals.length() == 0
+                            ? beforeEquals.substring(posNotEqual + 1)
+                            : beforeEquals.substring(posNotEqual + 1) + "=" + afterEquals;
+            leftHandSide = beforeEquals.substring(0, posNotEqual);
+            interiorlyNegated = true;
+        }
+        if (interiorlyNegated) {
+            final var complement = getNonNegatedPropertyQuerySet(leftHandSide, propertyPredicate);
+            result.addAll(complement.complement().removeAllStrings());
+        } else {
+            result.addAll(getNonNegatedPropertyQuerySet(leftHandSide, propertyPredicate));
+        }
+        return true;
+    }
+
+    private UnicodeSet getNonNegatedPropertyQuerySet(
+            String leftHandSide, String propertyPredicate) {
+        final var unqualifiedLeftHandSide = new StringBuilder(leftHandSide);
+        final var queriedVersion = parseVersionQualifier(unqualifiedLeftHandSide);
+        final var deducedQueriedVersion = queriedVersion == null ? implicitVersion : queriedVersion;
+
+        final var queriedProperties = IndexUnicodeProperties.make(deducedQueriedVersion);
+
+        if (propertyPredicate.length() == 0) {
+            // Either unary-property-query, or binary-property-query with an empty property-value.
+            final var script = queriedProperties.getProperty(UcdProperty.Script);
+            final var generalCategory = queriedProperties.getProperty(UcdProperty.General_Category);
+            if (script.isValidValue(unqualifiedLeftHandSide.toString())) {
```

Comment:
lots of .toString() calls here. please make a String after parsing the version out of leftHandSide. if there was no version prefix, then the leftHandSide itself is already the String you want.

probably easiest to do something like
```
if (there was a version prefix) {
    leftHandSide = unqualifiedLeftHandSide.toString();
}
... use leftHandSide ...
```

Reply:
Done, but kept the long name because I think I will need to look at the original leftHandSide in a future revision.